src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+    uint8_t mode      = 0;
+
+    // sensitivity
+    uint16_t senv_x_val       = 0;
+    uint16_t senv_y_val       = 0;
+    uint16_t dspeed_threshold = 0; // threshold, in mickeys/s  TODO: should
+                                   // affect mouse movement
+    float senv_x = 0.0f;
+    float senv_y = 0.0f;
+
+    // mouse position allowed range
+    int16_t minpos_x = 0;
+    int16_t maxpos_x = 0;
+    int16_t minpos_y = 0;
+    int16_t maxpos_y = 0;
+
+    // mouse cursor
+    uint8_t page                     = 0; // cursor display page number
+    bool inhibit_draw                = false;
+    uint16_t hidden                  = 0;
+    uint16_t oldhidden               = 0;
+    int16_t clipx                    = 0;
+    int16_t clipy                    = 0;
+    int16_t hot_x                    = 0; // cursor hot spot, horizontal
+    int16_t hot_y                    = 0; // cursor hot spot, vertical
+    bool background                  = false;
+    uint16_t backposx                = 0;
+    uint16_t backposy                = 0;
+    uint8_t backData[CURSOR_SIZE_XY] = {0};
+    MouseCursor cursor_type          = MouseCursor::Software;
+
+    // cursor shape definition
+    uint16_t text_and_mask                       = 0;
+    uint16_t text_xor_mask                       = 0;
+    bool user_screen_mask                        = false;
+    bool user_cursor_mask                        = false;
+    uint16_t user_def_screen_mask[CURSOR_SIZE_Y] = {0};
+    uint16_t user_def_cursor_mask[CURSOR_SIZE_Y] = {0};
+
+    // user callback
+    uint16_t sub_mask = 0;
+    uint16_t sub_seg  = 0;
+    uint16_t sub_ofs  = 0;
+
+} state;
+
+static RealPt uir_callback;
+
+// ***************************************************************************
+// Common helper routines
+// ***************************************************************************
+
+static uint16_t SignedToReg16(const int16_t x)
+{
+    if (x >= 0)
+        return static_cast<uint16_t>(x);
+    else
+        // -1 for 0xffff, -2 for 0xfffe, etc.
+        return static_cast<uint16_t>(0x10000 + x);
+}
+
+static uint16_t SignedToReg16(const float x)
+{
+    return SignedToReg16(static_cast<int16_t>(x));
+}
+
+static int16_t RegToSigned16(const uint16_t x)
+{
+    if (bit::is(x, b15))
+        // 0xffff for -1, 0xfffe for -2, etc.
+        return static_cast<int16_t>(x - 0x10000);
+    else
+        return static_cast<int16_t>(x);
+}
+
+static uint16_t GetPosX()
+{
+    return static_cast<uint16_t>(std::lround(pos_x)) & state.gran_x;
+}
+
+static uint16_t GetPosY()
+{
+    return static_cast<uint16_t>(std::lround(pos_y)) & state.gran_y;
+}
+
+// ***************************************************************************
+// Data - default cursor/mask
+// ***************************************************************************
+
+static constexpr uint16_t DEFAULT_TEXT_AND_MASK = 0x77FF;
+static constexpr uint16_t DEFAULT_TEXT_XOR_MASK = 0x7700;
```

Comment:
Avoid using upper case; I think it would be better to use lower_camel_case for these as well, no?